### PR TITLE
Implement Person, Student, and Teacher Classes

### PR DESCRIPTION
**Pull request by leloufadel**
This pull request addresses the following requirements:

**General Requirements**:

- [x] Ensured that there are no linter errors.
- [x] Followed Gitflow conventions for branch naming and workflow.
- [x] Documented the work following professional documentation standards.

**Ruby Requirements**:

- [x] Followed a list of best practices for Ruby coding.

**Project Requirements**:

- [x] Created a separate class file for each of the following classes: `Person`, `Student`, and `Teacher`.

**Details**:

- Implemented the `Person` class with the following features:
  - Instance variables: `@id`, `@name`, and `@age`.
  - Constructor with optional parameters for `name` (default: "Unknown"), `age`, and `parent_permission` (default: true).
  - Getters and setters for `@name` and `@age`.
  - Private method `of_age?` that returns true if `@age` is greater or equal to 18 and false otherwise.
  - Public method `can_use_services?` that returns true if the person is of age or has parental permission.

- Implemented the `Student` class, which inherits from `Person`, with the following features:
  - Extended the parent's constructor by adding `@classroom`.
  - Added a method `play_hooky` that returns "¯\\(ツ)/¯".

- Implemented the `Teacher` class, which also inherits from `Person`, with the following features:
  - Extended the parent's constructor by adding `@specialization`.
  - Overrode the `can_use_services?` method to always return true.

**Files Modified**:

- [x] Added `person.rb` containing the `Person` class.
- [x] Added `student.rb` containing the `Student` class.
- [x] Added `teacher.rb` containing the `Teacher` class.

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(classroom, id, age, name: 'Unknown', parent_permission: true)
+    super(id, age, name, parent_permission)
+    @classroom = classroom
+  end
+
```
Comment (Olamarx) [inline teacher.rb]: - [ ] You probably forgot or mispositioned the classroom parameter, it is supposed to be in the student while the specialization which is in the student is to be here. Kindly fix it up.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @leloufadel, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Commit messages meaningful ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 1 above)
Comment (leloufadel) [reply teacher.rb]: @Olamarx oh, I mispositioned them 😄 .

**Review by joel-kalema — APPROVED** [review]
Hi @leloufadel 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**